typescript/cli/src/tests/warp/warp-deploy.e2e-test.ts

```diff
@@ -0,0 +1,540 @@
+            currentOutput.includes('Please enter the private key for chain'),
+          input: `${ANVIL_KEY}${KeyBoardKeys.ENTER}`,
+        },
+      ];
+
+      // Deploy
+      const output = hyperlaneWarpDeployRaw({
+        warpCorePath: WARP_DEPLOY_OUTPUT_PATH,
+        skipConfirmationPrompts: true,
+      })
+        .stdio('pipe')
+        .nothrow();
+
+      const finalOutput = await handlePrompts(output, steps);
+
+      // Assertions
+      expect(finalOutput.exitCode).to.equal(0);
+      const updatedWarpDeployConfig_2 = await readWarpConfig(
+        CHAIN_NAME_2,
+        COMBINED_WARP_CORE_CONFIG_PATH,
+        WARP_DEPLOY_OUTPUT_PATH,
+      );
+
+      expect(updatedWarpDeployConfig_2[CHAIN_NAME_2].type).to.equal(
+        TokenType.collateral,
+      );
+      expect(updatedWarpDeployConfig_2[CHAIN_NAME_2].decimals).to.equal(
+        expectedTokenDecimals,
+      );
+      expect(updatedWarpDeployConfig_2[CHAIN_NAME_2].symbol).to.equal(
+        expectedTokenSymbol,
+      );
+      expect(updatedWarpDeployConfig_2[CHAIN_NAME_2].mailbox).to.equal(
+        chain2Addresses.mailbox,
+      );
+
+      const updatedWarpDeployConfig_3 = await readWarpConfig(
+        CHAIN_NAME_3,
+        COMBINED_WARP_CORE_CONFIG_PATH,
+        WARP_DEPLOY_OUTPUT_PATH,
+      );
+
+      expect(updatedWarpDeployConfig_3[CHAIN_NAME_3].type).to.equal(
+        TokenType.synthetic,
+      );
+      expect(updatedWarpDeployConfig_3[CHAIN_NAME_3].decimals).to.equal(
+        expectedTokenDecimals,
+      );
+      expect(updatedWarpDeployConfig_3[CHAIN_NAME_3].symbol).to.equal(
+        expectedTokenSymbol,
+      );
+      expect(updatedWarpDeployConfig_3[CHAIN_NAME_3].mailbox).to.equal(
+        chain3Addresses.mailbox,
+      );
+    });
+  });
+
+  describe(`hyperlane warp deploy --config ... --yes --key ...`, () => {
+    let token: ERC20Test;
+    let vault: ERC4626Test;
+
+    before(async () => {
+      token = await deployToken(ANVIL_KEY, CHAIN_NAME_2);
+      vault = await deploy4626Vault(ANVIL_KEY, CHAIN_NAME_2, token.address);
+    });
+
+    it('should only allow rebasing yield route to be deployed with rebasing synthetic', async function () {
+      const warpConfig: WarpRouteDeployConfig = {
+        [CHAIN_NAME_2]: {
+          type: TokenType.collateralVaultRebase,
+          token: vault.address,
+          mailbox: chain2Addresses.mailbox,
+          owner: chain2Addresses.mailbox,
+        },
+        [CHAIN_NAME_3]: {
+          type: TokenType.synthetic,
+          mailbox: chain2Addresses.mailbox,
+          owner: chain2Addresses.mailbox,
+        },
+      };
+
+      writeYamlOrJson(WARP_DEPLOY_OUTPUT_PATH, warpConfig);
+      await hyperlaneWarpDeploy(WARP_DEPLOY_OUTPUT_PATH).should.be.rejected; // TODO: revisit this to figure out how to parse the error.
+    });
+
+    it(`should be able to bridge between ${TokenType.collateralVaultRebase} and ${TokenType.syntheticRebase}`, async function () {
```

Comment:
could you define some sort of 2D matrix of token types and have a single tests logic test all of the combinations?